File: src/main/java/com/sat/auth/config/handler/CustomAuthenticationEntrypoint.java
package com.sat.auth.config.handler;

import com.fasterxml.jackson.databind.ObjectMapper;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.util.Map;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.web.AuthenticationEntryPoint;

public class CustomAuthenticationEntrypoint implements AuthenticationEntryPoint {

    private final ObjectMapper objectMapper = new ObjectMapper();

    @Override
    public void commence(HttpServletRequest request, HttpServletResponse response, AuthenticationException authException) throws IOException, ServletException {
        // TODO: ExceptionHandling 사용해 응답 규격 정의
        Map<String, String> errorMessages = Map.of("error", authException.getMessage());
        response.getWriter().write(objectMapper.writeValueAsString(errorMessages));
        response.setContentType("application/json");
        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
    }
}

File: src/main/kotlin/com/sat/common/config/security/CustomAuthenticationEntryPoint.kt
package com.sat.common.config.security

import com.fasterxml.jackson.databind.ObjectMapper
import com.sat.common.ui.web.ErrorResponse
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.http.HttpStatus
import org.springframework.security.core.AuthenticationException
import org.springframework.security.web.AuthenticationEntryPoint
import org.springframework.stereotype.Component

@Component
class CustomAuthenticationEntryPoint(
    private val objectMapper: ObjectMapper,
) : AuthenticationEntryPoint {
    override fun commence(
        request: HttpServletRequest,
        response: HttpServletResponse,
        authException: AuthenticationException
    ) {
        val writer = response.writer
        response.status = HttpStatus.UNAUTHORIZED.value()
        writer.println(objectMapper.writeValueAsString(ErrorResponse(authException.message)))
    }
}
